fs-base/src/main/java/com/palantir/giraffe/file/MoreFiles.java

```diff
@@ -493,71 +492,102 @@ public static boolean isEmpty(Path path) throws IOException {
-     * {@code Path} objects are obtained as if by resolving the name of the
-     * directory objects against the given {@code Path}.
+     * Gets a list of the files and directories in the given directory. The
+     * returned paths are {@linkplain Path#resolve(Path) resolved} against the
+     * given path.
      *
-     * @param directoryPath The {@code Path} to get the directory entries for
+     * @param dir the {@code Path} to list
      *
-     * @return A list of {@code Path} objects representing the directory entries
+     * @throws IOException if an I/O error occurs while listing entries
+     */
+    public static List<Path> listDirectory(Path dir) throws IOException {
+        try (DirectoryStream<Path> entries = Files.newDirectoryStream(dir)) {
+            return Lists.newArrayList(entries);
+        }
+    }
+
+    /**
+     * Gets a list of the files in the given directory. The returned paths are
+     * {@linkplain Path#resolve(Path) resolved} against the given path.
+     *
+     * @param dir the {@code Path} to list
      *
-     * @throws IOException If an I/O error occurs while accessing the path
+     * @throws IOException if an I/O error occurs while listing entries
      */
-    public static List<Path> listDirectory(Path directoryPath) throws IOException {
-        try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(directoryPath)) {
-            return Lists.newArrayList(directoryStream);
+    public static List<Path> listDirectoryFiles(Path dir) throws IOException {
+        try (DirectoryStream<Path> entries = Files.newDirectoryStream(dir, FILE_FILTER)) {
+            return Lists.newArrayList(entries);
         }
     }
 
+    private enum DirectoryListMode {
+        ALL, FILES_ONLY, DIRS_ONLY;
+    }
+
     /**
-     * A {@code FileVisitor} that builds a list of relative {@code Path} objects it visits.
+     * A {@code FileVisitor} that builds a list of {@code Path} objects it visits.
      *
      * @author jyu
      */
-    private static final class RelativeListVisitor extends SimpleFileVisitor<Path> {
-        private final Path root;
-        private final ImmutableList.Builder<Path> paths;
+    private static final class ListVisitor extends SimpleFileVisitor<Path> {
+        private final DirectoryListMode mode;
+        private final ArrayList<Path> paths;
+
+        ListVisitor(DirectoryListMode mode) {
+            this.mode = mode;
+            paths = new ArrayList<>();
+        }
 
-        public RelativeListVisitor(Path relativeRoot) {
-            root = Preconditions.checkNotNull(relativeRoot, "relative root must be non-null");
-            paths = new ImmutableList.Builder<>();
+        @Override
+        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
+                throws IOException {
+            if (mode == DirectoryListMode.ALL || mode == DirectoryListMode.DIRS_ONLY) {
```

Comment:
This kind of logic is indicative of awkward separation of list modes.

I would prefer to use a `Set<DirectoryListMode>` and provide a simple factories to construct the sets a la `filesOnly()`, `dirsOnly()`, and `all()`.

This way, this condition becomes `modes.contains(DirectoryListMode.DIRS)`.